proto-public/pbresource/resource.proto

```diff
@@ -469,6 +469,10 @@ message WatchListRequest {
   // NamePrefix filters the results to those with a name beginning with the
   // given prefix.
   string name_prefix = 3;
+
+  // SendSnapshotOperations opts into receiving the pair of
+  // OPERATION_START_OF_SNAPSHOT and OPERATION_END_OF_SNAPSHOT.
+  bool send_snapshot_operations = 4;
```

Comment:
I made this conditional for two reasons (both of which I don't hold dearly):
- back compat with prior versions (only vaguely important pre-1.18)
- perhaps most callers would not care about the snapshot framing?